examples/dictionary-example.php

```diff
@@ -1,28 +1,49 @@
 <?php
+declare(strict_types=1);
 
 require "vendor/autoload.php";
 
 use Momento\Auth\EnvMomentoTokenProvider;
 use Momento\Cache\SimpleCacheClient;
+use Momento\Config\Configurations\Laptop;
+use Monolog\Formatter\LineFormatter;
+use Monolog\Handler\StreamHandler;
+use Monolog\Logger;
+use Psr\Log\LoggerInterface;
 
-$MOMENTO_AUTH_TOKEN = getenv("MOMENTO_AUTH_TOKEN");
 $CACHE_NAME = getenv("CACHE_NAME");
 $DICTIONARY_NAME = "example-dictionary";
 $FIELD = "MyField";
 $VALUE = "MyValue";
 $ITEM_DEFAULT_TTL_SECONDS = 60;
 
-function printBanner(string $message): void
+// Setup
+$authProvider = new EnvMomentoTokenProvider("MOMENTO_AUTH_TOKEN");
+
+// Use your favorite PSR-3 logger.
+$logger = new Logger("example");
```

Comment:
I like what you are doing here but just one question - does PSR-3 specify an interface for a logger *factory* too?  Or just for a logger?

The reason I ask is that in most languages the logging frameworks will allow/encourage you to instantiate a logger scoped to a specific class or module, so that each log message can clearly identify where it came from.  With the approach you have here we sort of assume a fairly global/monolithic logger that won't clearly distinguish which log messages come from us vs come from other sources.  Am I reading that right?

It's also entirely possible that that granular logging concept isn't common in PHP and that this is indeed the idiomatic / best-practice way to do this, but I wanted to ask.

Reply:
There isn't a concept of factories in [PSR-3](https://www.php-fig.org/psr/psr-3/). In the [Monolog](https://github.com/Seldaek/monolog) example above, the string "example" in the constructor is the logging channel name, and that channel name can be included in log messages if desired. Monolog can instantiate multiple loggers with multiple channel names and point each one to different output stream or service with custom formats. It's very configurable, kind of to the point where I'm not sure it's possible to expose a single factory capable of instantiating objects from any arbitrary PSR-3 implementation.

That said, I'll look into some of the other PSR-3 implementations and see if there is any commonality I can use to provide some kind of factory.

Reply:
Yeah, here's a taste of the diversity of logging approaches: https://www.loggly.com/ultimate-guide/php-logging-libraries/. I don't think the factory approach is feasible in this ecosystem unless we choose/bless a specific logging solution and add a factory for it, which defeats the purpose of generalizing to PSR-3 in the first place.

Reply:
okay let's merge this.  i did have one other thought i'd like to chat with you about for 5 mins but it doesn't block this PR.